apps/api/src/domain/models/song/commands/add-song.command.integration.spec.ts

```diff
@@ -0,0 +1,157 @@
+import TestRepositoryProvider from '../../../../persistence/repositories/__tests__/TestRepositoryProvider';
+import { DTO } from '../../../../types/DTO';
+import InvalidEntityDTOError from '../../../domainModelValidators/errors/InvalidEntityDTOError';
+import MissingSongTitleError from '../../../domainModelValidators/errors/song/MissingSongTitleError';
+import getValidEntityInstaceForTest from '../../../domainModelValidators/__tests__/domainModelValidators/utilities/getValidEntityInstaceForTest';
+import { ResourceType } from '../../../types/ResourceType';
+import buildInMemorySnapshot from '../../../utilities/buildInMemorySnapshot';
+import InvalidCommandPayloadTypeError from '../../shared/common-command-errors/InvalidCommandPayloadTypeError';
+import { AddSong } from './add-song.command';
+import { AddSongHandler } from './add-song.command-handler';
+
+export const dummyUuid = '9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d';
+
+const addSongCommandType = 'ADD_SONG';
+
+const validCommandFSA: FluxStandardAction<DTO<AddSong>> = {
+    type: addSongCommandType,
+    payload: {
+        id: dummyUuid,
+        title: 'test-song-name (language)',
+        titleEnglish: 'test-song-name (English)',
+        contributorAndRoles: [],
+        lyrics: 'la la la',
+        audioURL: 'https://www.mysound.org/song.mp3',
+        lengthMilliseconds: 15340,
+    },
+};
+
+const validPayload = validCommandFSA.payload;
+describe('AddSong', () => {
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let arangoConnectionProvider: ArangoConnectionProvider;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, arangoConnectionProvider } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: generateRandomTestDatabaseName(),
+            }));
+
+        commandHandlerService.registerHandler(
+            addSongCommandType,
+            new AddSongHandler(testRepositoryProvider)
+        );
+    });
+
+    afterAll(async () => {
+        await arangoConnectionProvider.dropDatabaseIfExists();
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    afterEach(async () => {
+        await testRepositoryProvider.testTeardown();
+    });
+
+    describe('when the payload is valid', () => {
+        it('should return ok', async () => {
+            await testRepositoryProvider.addFullSnapshot({
+                resources: {
+                    song: [],
+                },
+                tags: [],
+                connections: [],
+                categoryTree: [],
+            });
+
+            const result = await commandHandlerService.execute(validCommandFSA);
+
+            expect(result).toBe(Ack);
+        });
+    });
+
+    describe('when the payload has an invalid type', () => {
```

Comment:
I think it is time to work on [this story[(https://www.pivotaltracker.com/story/show/182217249), which will give us a way to generate payloads that violate the type rules.